ffi/src/schema.rs

```diff
@@ -143,21 +147,21 @@ pub unsafe extern "C" fn visit_schema(
     fn visit_struct_fields(visitor: &EngineSchemaVisitor, s: &StructType) -> usize {
         let child_list_id = (visitor.make_field_list)(visitor.data, s.fields.len());
         for field in s.fields() {
-            visit_schema_item(field.data_type(), field.name(), visitor, child_list_id);
+            visit_schema_item(field.data_type(), field.name(), visitor, child_list_id, field.is_nullable());
```

Comment:
The `visit_schema_item` helper is private, so we should consider where to inject the new arg for maximum readability going forward -- probably _not_ at end like this?

Honestly, the current arg order is already a bit weird, so maybe we should reorder the whole thing:
```suggestion
            visit_schema_item(field.name(), field.data_type(), field.is_nullable(), visitor, child_list_id);
```

Reply:
yea I agree if it isn't too much trouble to just reorder the args like proposed here - it matches the typical parameter ordering that i'm aware of in Arrow, etc.